### ISSUE TITLE
OPEN API or Take Function

### ISSUE THREAD

**Issue by eledtech1234** [bitfocus/companion-module-novastar-h-series#3]
Hi, 
is that possible somebody can help me or making OPENAPI Module for Novastar H Series?
in attachment the documentation.
I would at least need the TAKE Function.
thanks in advance.

best erik
[H系列对接OpenApi.pdf](https://github.com/bitfocus/companion-module-novastar-h-series/files/9894358/H.OpenApi.pdf)

**Comment by eledtech1234** [comment]
TAKE Function UDP:
[{"cmd":"W0412","deviceId":0,"screenId":0,"enNonTime": 1, "manualPlay": 1}]

LIVE MODE ON UDP:
[{"cmd":"W0412","deviceId":0,"screenId":0,"enNonTime": 0, "manualPlay": 0}]

LIVE MODE OFF UDP:
[{"cmd":"W0412","deviceId":0,"screenId":0,"enNonTime": 1, "manualPlay": 0}]

these are working.
Unfortnatly over UDP Webbrowser Tool doesnt update, when switch preset. 
Over HTTP Open API it seems to work, at least when you work with 2 PCs and remote the H Machine, it will update on both PCs as soon you switch Preset or change anything.

**Comment by rit3sh7** [comment]
I will try this but I dont think it will be soon. Super busy with work till
end of November unfortunately. Sorry this won't be quick add. I can
possibly make custom build for you faster then uploading this to main repo
once i try this out.

**Comment by eledtech1234** [comment]
hi, thanks for replying.
for now its working, as you implent "send commands"
so no rush :)

**Comment by alferca4** [comment]
Hi, how you calculate the cmd number? For example th W0412 The API documentation is in Chinese and I can’t found it

**Comment by eledtech1234** [comment]
it was trial and error. 
You see in Document File Article 4.13 (layer/layerscreentake), i tried first with CMD 0413 and didnt work.
Then i just tried -1 = 0412 and it worked.
i tried with more articles and reduced by 1, which seems to work in all cases.
Sorry for my english

**Comment by rit3sh7** [comment]
Yes its off by 1 as in code it starts with 0 but we normally count staring with #1 so I’ve had to do that in code so people don’t confuse it. Such as Preset 1 is technically 0 according to API calls.
